ops/pebble.py

```diff
@@ -2029,7 +2030,7 @@ def _wait_change_using_wait(self, change_id: ChangeID, timeout: Optional[float])
 
             try:
                 return self._wait_change(change_id, this_timeout)
-            except TimeoutError:
+            except (socket.timeout, builtins.TimeoutError):
```

Comment:
People looking at this might wonder why we have both, so let's add a comment, something like:

```suggestion
            except (socket.timeout, builtins.TimeoutError):
                # NOTE: in Python 3.10 socket.timeout is an alias of TimeoutError,
                # but we still need to support Python 3.8, so catch both.
```